src/CoreWCF.WebHttp/src/CoreWCF/Description/WebHttpServiceModelCompat.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using CoreWCF.Collections.Generic;
+using CoreWCF.Web;
+
+namespace CoreWCF.Description;
+
+/// <summary>
+/// This class contains the logic to convert System.ServiceModel.Web attributes into
+/// CoreWCF.WebHttp attributes.
+/// This allows endpoints to be defined using WCF classic attributes as well as CoreWCF
+/// attributes, for a simpler migration path.
+/// </summary>
+internal static class WebHttpServiceModelCompat
+{
+
+    public static void ServiceModelAttributeFixup(ServiceEndpoint endpoint)
+    {
+        foreach (OperationDescription operationDescription in endpoint.Contract.Operations)
+        {
+            if (operationDescription.OperationBehaviors is KeyedByTypeCollection<IOperationBehavior> behaviors
+                && behaviors.Find<WebGetAttribute>() == null
+                && behaviors.Find<WebInvokeAttribute>() == null)
+            {
+                CheckForAndConvertSmAttributes(operationDescription);
+            }
+        }
+    }
+
+    private static void CheckForAndConvertSmAttributes(OperationDescription od)
+    {
+        var opMethod = od.SyncMethod ?? od.TaskMethod ?? od.BeginMethod;
+        var attributes = opMethod.GetCustomAttributes().ToArray();
+
+        var convertedAttributes = AttributeConverters.Convert(attributes);
+        foreach(var converted in convertedAttributes)
+            od.OperationBehaviors.Add(converted);
+
+    }
+
+    private static class AttributeConverters
+    {
+
+        private static readonly IReadOnlyDictionary<string, IAttributeConverter> s_attributeConverters =
+            new Dictionary<string, IAttributeConverter>
+            {
+                { "System.ServiceModel.Web.WebGetAttribute", new AttributeConverter<WebGetAttribute>() },
+                { "System.ServiceModel.Web.WebInvokeAttribute", new AttributeConverter<WebInvokeAttribute>() }
+            };
+
+        public static IEnumerable<IOperationBehavior> Convert(IEnumerable<Attribute> attributes)
+        {
+            foreach (Attribute attribute in attributes)
+            {
+                if (Convert(attribute) is { } convertedAttribute)
+                    yield return convertedAttribute;
```

Comment:
I've never seen this syntax before. What does `is { }` do?

Reply:
That is the new(ish) C#9 conditional pattern matching.

The full form is this.

`if (rhs is SomeType { patternMatch } lhs) { }`

We can omit the `SomeType` because we are casting from `IOperationBehavior?` to `IOperationBehavior`. 

The pattern we are matching is empty (ie, match on any).

Overall, this is just a nice way to do a null check with the new nullable types, whilst putting the value into a variable that is explicitly not null.